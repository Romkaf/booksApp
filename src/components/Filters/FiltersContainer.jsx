import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
	setAuthorFilter,
	setTitleFilter,
	setCurrentPage,
} from '@models/actions';
import Filters from './Filters';

const FiltersContainer = () => {
	const dispatch = useDispatch();
	const { titleFilter, authorFilter } = useSelector((state) => state.filters);

	const handleFilterSet = (handler) => (value) => {
		dispatch(handler(value));
		dispatch(setCurrentPage(1));
	};

	const filters = [
		{
			title: 'Title',
			handler: handleFilterSet(setTitleFilter),
			value: titleFilter,
		},
		{
			title: 'Author',
			handler: handleFilterSet(setAuthorFilter),
			value: authorFilter,
		},
	];

	return <Filters filters={filters} />;
};

export default FiltersContainer;
