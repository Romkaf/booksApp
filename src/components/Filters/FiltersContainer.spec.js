import React from 'react';
import FiltersContainer from './FiltersContainer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

describe('FiltersContainer', () => {
	const mockStore = configureMockStore([thunk]);

	const store = mockStore({
		filters: { titleFilter: '', authorFilter: '' },
	});

	const wrapper = mount(
		<Provider store={store}>
			<FiltersContainer />
		</Provider>,
	);

	it('should render component', () => {
		const component = wrapper.find('FiltersContainer');
		expect(component).toMatchSnapshot();
	});
});
