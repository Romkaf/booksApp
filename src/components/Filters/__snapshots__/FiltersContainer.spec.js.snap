// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersContainer should render component 1`] = `
<FiltersContainer>
  <Filters
    filters={
      Array [
        Object {
          "handler": [Function],
          "title": "Title",
          "value": "",
        },
        Object {
          "handler": [Function],
          "title": "Author",
          "value": "",
        },
      ]
    }
  >
    <div
      className="root"
    >
      <h2
        className="heading"
      >
        Filters
      </h2>
      <ul
        className="filtersWrapper"
      >
        <li
          key="Title"
        >
          <Filter
            label="Title"
            onChange={[Function]}
            value=""
          >
            <div
              className="field"
            >
              <input
                className="input"
                id="Title"
                onChange={[Function]}
                type="text"
              />
              <label
                className="label"
                htmlFor="Title"
              >
                Title
              </label>
            </div>
          </Filter>
        </li>
        <li
          key="Author"
        >
          <Filter
            label="Author"
            onChange={[Function]}
            value=""
          >
            <div
              className="field"
            >
              <input
                className="input"
                id="Author"
                onChange={[Function]}
                type="text"
              />
              <label
                className="label"
                htmlFor="Author"
              >
                Author
              </label>
            </div>
          </Filter>
        </li>
      </ul>
    </div>
  </Filters>
</FiltersContainer>
`;
